ui/app/models/secret-v2.js

```diff
@@ -8,7 +8,7 @@ import { withModelValidations } from 'vault/decorators/model-validations';
 
 const validations = {
   maxVersions: [
-    { type: 'number', options: { asString: true }, message: 'Maximum versions must be a number.' },
```

Comment:
This seems a little confusing because originally the `defaultValue` is an integer, and then after interacting with the form it becomes a string.

Reply:
Yeah - this was part of the original bug actually. `maxVersions` is returned as an integer from the API and defined on the model as a number (in the model's attrs) but when interacting with the form (when validations happen) it's a string. 

![max-versions](https://user-images.githubusercontent.com/68122737/170315255-1c0f9fdf-0901-49da-8949-cbf82aded0ef.gif)

Reply:
Thanks. So I think if we are not concerned with validating type in the number validator than it is fine to take out the `asString` option.